### PR TITLE
Not generating assembly versions if it already exists

### PR DESCRIPTION
**Pull request by otawfik-ms**
Saves msbuild rebuild time.

### CODE REVIEW

**Comment by otawfik-ms** [comment]
@Microsoft/fsharp-compiler

**Comment by dsyme** [comment]
Yes!!! Thank you!!!!

File: src/Microbuild.Settings.targets [hunk 1]
```diff
@@ -155,9 +156,18 @@
                 throw new ArgumentException("Language " + Language + " is not supported");
         }
         
-        AssemblyFileVersionPath = Path.GetTempFileName() + "." + Language;
-        Log.LogMessage("Generating AssemblyFileVersion file: " + AssemblyFileVersionPath, MessageImportance.High);
-        File.WriteAllText(AssemblyFileVersionPath, stringBuilder.ToString());
+        var fileContents = stringBuilder.ToString();
+        AssemblyFileVersionPath = Path.Combine(Path.GetTempPath(), string.Join(".", AssemblyName, "AssemblyFileVersion", Language));
+        
+        if (File.Exists(AssemblyFileVersionPath) && File.ReadAllText(AssemblyFileVersionPath) == fileContents)
```
Comment (jaredpar) [inline src/Microbuild.Settings.targets]: What is the MSBuild behavior here if `ReadAllText` throws an exception?  Have to assume I/O can and will fail at some point.

Comment (otawfik-ms) [inline src/Microbuild.Settings.targets]: If there is a temporary I/O issue the target would fail. Do you use a similar mechanism in Roslyn?

Comment (jaredpar) [inline src/Microbuild.Settings.targets]: I don't think we do any direct I/O in the targets / settings files.  I doubt this would cause an issue though.

**Comment by jaredpar** [comment]
:+1:

**Pull request merged**